kong/plugins/oauth2/access.lua

```diff
@@ -450,19 +450,21 @@ end
 
 local function parse_access_token(conf)
   local found_in = {}
-  local result = retrieve_parameters()["access_token"]
-  if not result then
-    local authorization = ngx.req.get_headers()[conf.auth_header_name]
-    if authorization then
-      local parts = {}
-      for v in authorization:gmatch("%S+") do -- Split by space
-        table.insert(parts, v)
-      end
-      if #parts == 2 and (parts[1]:lower() == "token" or parts[1]:lower() == "bearer") then
-        result = parts[2]
-        found_in.authorization_header = true
-      end
+  local access_token = ngx.req.get_headers()[conf.auth_header_name]
+  if access_token then
+    if type(access_token) == "table" then --Take the first found
+      access_token = access_token[1]
```

Comment:
This sort of change (which is a great addition!) would require a test to protect ourselves from future regressions.

Reply:
Thought this might pop up as I was thinking about that latest enhancement. Would a test showing how a request containing 2 of the same correct Bearer token headers yielding success be enough? Something similar to the lines of https://github.com/Kong/kong/blob/master/spec/03-plugins/26-oauth2/03-access_spec.lua#L1334 
```
        it("returns success with multiple custom duplicate authorization headers", function()
          local res = assert(proxy_ssl_client:send {
            method  = "POST",
            path    = "/oauth2/token",
            body    = {
              provision_key        = "provision123",
              authenticated_userid = "id123",
              scope                = "email",
              grant_type           = "password"
            },
            headers = {
              ["Host"]             = "oauth2_5.com",
              ["Content-Type"]     = "application/json",
              CustomAuthHeader = "Basic Y2xpZW50aWQxMjM6c2VjcmV0MTIz",
              CustomAuthHeader = "Basic Y2xpZW50aWQxMjM6c2VjcmV0MTIz"
            }
          })
          local body = assert.res_status(200, res)
          assert.is_table(ngx.re.match(body, [[^\{"refresh_token":"[\w]{32,32}","token_type":"bearer","access_token":"[\w]{32,32}","expires_in":5\}$]]))
        end)
```

Since we know nginx handles dup Authorization headers itself only time the code prior would have failed would be on a declared custom Authorization header in the oauth2 plugin config.

Reply:
@jeremyjpj0916 Yes, this is the test I had in mind. However, in order to specify the same header multiple times, you would have to specify it as such:
```lua
headers = {
  ["Header-Name"] = { "val1", "val2" },
}
```

Reply:
Feeling this snippit:
```
      it("works when a correct access_token is being sent in duplicate custom headers", function()
        local token = provision_token("oauth2_11.com",nil,"clientid1011","secret1011")

        local res = assert(proxy_ssl_client:send {
          method = "GET",
          path = "/request",
          headers = {
            ["Host"] = "oauth2_11.com",
            ["custom_header_name"] = { "bearer " .. token.access_token, "bearer " .. token.access_token },
          }
        })
        assert.res_status(200, res)
      end)
```